### DISCUSSION TITLE
Emergency limit

### DISCUSSION THREAD

**Discussion by JensHauser** [spali/go-rscp#53] [Q&A]
I´m interested to set the emergency limits (e.g. max. unloading down to 2kWh) in my S10X.

Couldn´t get it done so far.
In this and other forums I [found](https://github.com/nischram/E3dcGui/issues/58#issuecomment-930448160) hints in relation to this subject, maybe helpful?

* TAG_SE_REQ_EP_RESERVE                                 0x1B000009
* TAG_SE_EP_RESERVE                                     0x1B800009
* EMS_REQ_SET_EMERGENCY_POWER Tag = 0x01000074
* EMS_SET_EMERGENCY_POWER = 0x01800074
* ISE_HM_SEND_EMERGENCY_POWER_SIZE
* TAG_EMS_ISE_EMERGENCY_POWER_SIZE
* "TagName": "REQ_SET_EMERGENCY_POWER", "TagNameGlobal": "TAG_EMS_REQ_SET_EMERGENCY_POWER",
* "TagName": "SET_EMERGENCY_POWER",		"TagNameGlobal": "TAG_EMS_SET_EMERGENCY_POWER",

None of them work for me in that way:
`./e3dc '[["TAG_EMS_ISE_EMERGENCY_POWER_SIZE"]]'`
--> error: TAG_EMS_ISE_EP_SIZE does not belong to Tag values

Also found a list of tags from the E3DC [website](https://s10.e3dc.com/s10/js/rscpLibV0.9.3.min.js) which might be related to emergency power
* EMS_REQ_EP_RESERVE
* EMS_REQ_SET_TOTAL_EP_RESERVE_W
* EMS_EP_RESERVE
* EMS_PARAM_FULL_CHARGED_ENERGY_EP_RESERVE
* EMS_PARAM_CURR_CHARGED_ENERGY_EP_RESERVE
* EMS_SET_TOTAL_EP_RESERVE_W
* SE_REQ_EP_RESERVE
* SE_REQ_SET_EP_RESERVE
* SE_PARAM_EP_RESERVE
* SE_PARAM_EP_RESERVE_MAX_W
* SE_PARAM_EP_RESERVE_W
* SE_EP_RESERVE

As I´m quite new in using this cool tool, maybe somehow has an advice?

**Comment by hmichels** [comment]
Hello!
With discussion #45 I've the same wish to use/set the emergency values on my S10X 😉.

The problem is that the necessary tags (listed above) - and maybe the correct using of these tags in the correct form (container) - are not defined in the current rscp library.
Unfortunately, I have no idea whether we can simply include the tags in the source code or not. However, the project still needs to be recompiled to create a new library.

We can only hope that Spali will help us here.
